lib/msal-node/docs/errors.md

```diff
@@ -0,0 +1,171 @@
+
+Ensure that the authority provided matches the authority in your azure app registration.
+```javascript
+const config = {
+    auth: {
+        authority: "https://login.microsoftonline.com/<provide-authority-that-matches-your-app-registration>"
+        // ...   
+    }
+}
+```
+
+2. When there is no internet connectivity. Details: ClientConfigurationError: `untrusted_authority`: The provided authority is not a trusted authority. Please include this authority in the knownAuthorities config parameter. It is also thrown when no authority is appended to the authority endpoint.
+    
+
+### Invalid_assertion
+
+**Error Message**: Client assertion must meet requirements described in https://tools.ietf.org/html/rfc7515. 
+
+This error is thrown when either the certificate thumbprint or public key is not provided or commented out.
+
+❌ The following example will throw this error because `thumbprint` is not provided:
+
+```javascript
+auth: {
+        //...
+        clientCertificate: {
+            thumbprint: "", // can be obtained when uploading certificate to Azure AD
+            privateKey: privateKey,
+        }
+    }
+```
+
+❌ The following example will throw this error because `publicKey` is not provided:
+
+```javascript
+auth: {
+        //...
+        clientCertificate: {
+            thumbprint: "YOUR_THUMBPRINT", // can be obtained when uploading certificate to Azure AD
+            privateKey: "",
+        }
+    }
+```
+
+
+### Request_cannot_be_made
+
+**Error message**: Token request cannot be made without authorization code or refresh token.
+
+This error is thrown when no authority is provided for auth flows such as auth-code-with-certs. 
+
+❌ The following example will throw this error because `authority` is commented out:
+
+```javascript
+auth: {
+        clientId: "CLIENT_ID",
+        // authority: "https://login.microsoftonline.com/YOUR-AUTHORITY",
+        clientCertificate: {
+            thumbprint: "YOUR_THUMBPRINT", 
+            privateKey: privateKey,
+        }
+    }
+```
+
+
+### Invalid_client_credential
+
+**Error message**: Client credential (secret, certificate, or assertion) must not be empty when creating a confidential client. An application should at most have one credential
+
+This error is thrown when the client's certificate is not provided.
+
+❌ The following example will throw this error because `clientCertificate` is commented out:
+
+```javascript
+auth: {
+        clientId: "CLIENT_ID",
+        authority: "https://login.microsoftonline.com/YOUR-AUTHORITY",
+        // clientCertificate: {
+        //     thumbprint: "YOUR_THUMBPRINT", 
+        //     privateKey: privateKey,
+        // }
+    }
+```
+
+
+## ClientConfigurationErrors
+
+### Empty_url_error
+
+**Error message**: URL was empty or null.
+
+This Error is thrown when no authority endpoint is provided in flows such as auth-code-with-certs.
```

Comment:
The library should default to `https://login.microsoftonline.com/common` if no authority is provided. This error should never be thrown unless there's a bug in the library.